notebook/agentchat_sql.ipynb

```diff
@@ -0,0 +1,858 @@
+    "    a. We still recommend increasing the content size as much as possible, as more information can be significantly beneficial.\n",
+    "2. I provided a customized (real-world) database in the example, because I found that GPT is already pretrained on many public databases, such as IMDB, TPC-H, and Chinook. The provided database is from Azure Brain project, with real tables and columns. Names and values in the rows are randomly anonymized for privacy considerations. The simple database with 2 tables can show a minimum example of SQLAgent for education purpose.\n",
+    "3. We initiate the SQLAgent with a warm start by default, which loads some example questions and answers into the history (i.e., oai_messages).\n",
+    "\n",
+    "\n",
+    "TODOs:\n",
+    "A flag check which column schema the agent has seen before.\n",
+    "\n",
+    "\n",
+    "Acknowledgement:\n",
+    "This notebook is inspired by LangChain's SQL Chain, Microsoft's DB CoPilot, and Microsoft's C&AI Team. We would also like to thank the Azure Allocator Brain team."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ea21a3f",
+   "metadata": {},
+   "source": [
+    "### Before everything starts, install AutoGen\n",
+    "```bash\n",
+    "pip install \"pyautogen~=0.2.0b4\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "e971a798",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sqlite3\n",
+    "import autogen\n",
+    "import requests\n",
+    "import json\n",
+    "import pdb\n",
+    "import os\n",
+    "import re\n",
+    "import tiktoken\n",
+    "\n",
+    "\n",
+    "from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union\n",
+    "\n",
+    "import autogen\n",
+    "from autogen import AssistantAgent, Agent, UserProxyAgent, ConversableAgent\n",
+    "from autogen.code_utils import DEFAULT_MODEL, UNKNOWN, content_str, execute_code, extract_code, infer_lang\n",
+    "\n",
+    "from termcolor import colored\n",
+    "import random\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "52eaceef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config_list_gpt4 = autogen.config_list_from_json(\n",
+    "    \"OAI_CONFIG_LIST\",\n",
+    "    filter_dict={\n",
+    "        \"model\": [\"gpt-4\", \"gpt-4-0314\", \"gpt4\", \"gpt-4-32k\", \"gpt-4-32k-0314\", \"gpt-4-32k-v0314\"],\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "gpt4_llm_config = {\"config_list\": config_list_gpt4, \"cache_seed\": 42}\n",
+    "\n",
+    "\n",
+    "\n",
+    "config_list_gpt35 = autogen.config_list_from_json(\n",
+    "    \"OAI_CONFIG_LIST\",\n",
+    "    filter_dict={\n",
+    "        \"model\": [\"gpt-35-turbo\", \"gpt-3.5-turbo\"],\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "gpt35_llm_config = {\"config_list\": config_list_gpt35, \"cache_seed\": 42}\n"
+   ]
+  },
+  {
```

Comment:
There're quite a few empty cells in the Notebook

![image](https://github.com/microsoft/autogen/assets/7947027/78b4954b-2eec-45e6-aef0-9c78d5514ef2)